tests/rptest/tests/cluster_view_test.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2020 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import random
+import time
+
+import requests
+import json
+from ducktape.mark import ignore
+from ducktape.mark.resource import cluster
+from ducktape.utils.util import wait_until
+from rptest.clients.kafka_cat import KafkaCat
+from rptest.clients.types import TopicSpec
+from rptest.services.admin import Admin
+from rptest.services.redpanda import RedpandaService
+from rptest.tests.end_to_end import EndToEndTest
+
+
+class ClusterViewTest(EndToEndTest):
+    @cluster(num_nodes=3)
+    def test_view_changes_on_add(self):
+        self.redpanda = RedpandaService(self.test_context, 3)
+        # start single node cluster
+        self.redpanda.start(nodes=[self.redpanda.nodes[0]])
+
+        admin = Admin(self.redpanda)
+
+        seed = None
+
+        def rp1_started():
```

Comment:
no need to change this, but there is now a helper called `utils.wait_until_result` which implements the pattern of combining wait_until with returning a value :)

Reply:
https://github.com/vectorizedio/redpanda/blob/dev/tests/rptest/util.py#L46

Reply:
interesting, thanks!